src/viking.js

```diff
@@ -1,15 +1,122 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health=health;
+    this.strength=strength;
+    
+  } // -- constructor
+attack (){
+  return this.strength
+}
+receiveDamage(damage){
+  this.health = this.health - damage
+  return 
+}
+} // -- class
+// let damage
+const instancia = new Soldier(10, 22);
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength){
+    super(health, strength);
+    this.name=name;
+  }
+  battleCry(){
+    return "Odin Owns You All!"
+  }
+  receiveDamage(damage){
+    this.health = this.health - damage
+
+    if (this.health>0){
+      return `${this.name} has received ${damage} points of damage`
+    } else {
+      return `${this.name} has died in act of combat`
+
+    } 
+  }
+  
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  constructor(health, strength){
+    super(health, strength);
+  }
+  
+  receiveDamage(damage){
+    this.health = this.health - damage
+    if (this.health>0){
+      return `A Saxon has received ${damage} points of damage`
+    } else {
+      return "A Saxon has died in combat"
+      
+    } 
+  }
+}
+
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = []
+    this.saxonArmy = []
+  }
+  
+  addViking(viking){
+    this.vikingArmy.push(viking)   
+    return 
+    
+  }
+  addSaxon(saxon){
+    this.saxonArmy.push(saxon)
+    return 
+    
+  }
+  //---------------------------------
+  vikingAttack(){
+    //Saxon.health=Viking.strength
```

Comment:
El resto de las iteraciones se pueden resolver usando un [Math.floor(Math.random() * this.saxonArmy.length)] igual se implementa para el vikingo, intentalo y me platicas que tal!!